src/lstm/recodebeam.cpp

```diff
@@ -512,8 +512,10 @@ void RecodeBeamSearch::ContinueContext(const RecodeNode* prev, int index,
            (previous->duplicate || previous->code == null_char_)) {
       previous = previous->prev;
     }
-    prefix.Set(p, previous->code);
-    full_code.Set(p, previous->code);
+    if (previous != nullptr) {
```

Comment:
Do you think that `previous == nullptr` would be possible and valid here? If not, adding `ASSERT_HOST(previous != nullptr);` might be better.

Reply:
i think so, the while loop just checks if the current node in the chain is not a nullpointer. If the next linked member is a nullpointer, it is assigned to previous and the loop ends.